addons/account/tests/test_payment_term.py

```diff
@@ -403,3 +403,40 @@ def test_payment_term_percent_round_calculation(self):
                 Command.create({'value_amount': 16.8, 'value': 'percent', 'nb_days': 0, }),
             ],
         })
+
+    def test_payment_term_labels(self):
+        # create a payment term with 40% now, 30% in 30 days and 30% in 60 days
+        multiple_installment_term = self.env['account.payment.term'].create({
+            'name': "test_payment_term_labels",
+            'line_ids': [
+                Command.create({'value_amount': 40, 'value': 'percent', 'nb_days': 0, }),
+                Command.create({'value_amount': 30, 'value': 'percent', 'nb_days': 30, }),
+                Command.create({'value_amount': 30, 'value': 'percent', 'nb_days': 60, }),
+            ],
+        })
+        # create an invoice with standard payment term
+        invoice = self.init_invoice('out_invoice', products=self.product_a)
+        # create immediate payment term
+        immediate_term = self.env['account.payment.term'].create({
+            'name': 'Immediate',
+            'line_ids': [
+                Command.create({'value_amount': 100, 'value': 'percent', 'nb_days': 0, }),
+            ],
+        })
+        invoice.invoice_payment_term_id = immediate_term
+        # post the invoice
+        invoice.action_post()
```

Comment:
No need to post and rest to draft.
The lines should be created before posting

Reply:
😭 new issue found, if I don't post the compute don't get triggered which causes some problems, making the compute depend on `invoice_payment_term_id` doesn't fix the problem either, what do you think would be a better solution ?
putting someting else in the compute depends ?

Reply:
Why doesn't it solve the issue?
Even by using this? 
https://github.com/odoo/odoo/pull/147670#discussion_r1440484375

Reply:
ok, adding `needed_terms` in the dependency array of the compute fixed the problem, but i don't know if it's optimal, let me know !

Reply:
yes, sorry, adding that did the job, Thanks boss !